src/main/java/com/datastax/cdm/schema/CqlTable.java

```diff
@@ -564,4 +591,101 @@ protected static ConsistencyLevel mapToConsistencyLevel(String level) {
         return retVal;
     }
 
+    /**
+     * Checks if the specified keyspace has token overlap issues by querying the system.size_estimates table.
+     *
+     * @param cqlSession
+     *            The CQL session to use for executing the query
+     * @param keyspaceName
+     *            The name of the keyspace to check
+     *
+     * @return true if token overlap is detected, false otherwise
+     */
+    private boolean hasTokenOverlap(CqlSession cqlSession, String keyspaceName) {
+        // Return false if either the session or keyspace name is null
+        if (cqlSession == null || keyspaceName == null || keyspaceName.isEmpty()) {
+            return false;
+        }
+
+        try {
+            // Execute query to check for token ranges for the specific keyspace
+            String query = "SELECT start_token, end_token FROM system.size_estimates WHERE keyspace_name = ?";
```

Comment:
We will have to evaluate if this is same across different flavors of C* engines, e.g. OSS C*, DSE, HCD, Astra  DB, etc.,